### PR TITLE
PF396 Fix erreur connection avec numero fiscal se terminant par une lettre

### PR DESCRIPTION
**Pull request by qcattez**
connect

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -47,10 +47,10 @@ def deconnexion
 
 private
   def param_numero_fiscal
-    params[:numero_fiscal].try(:delete, ' ')
+    params[:numero_fiscal].try(:delete, ' ').gsub(/\D+/, '')
```
Comment (kemenaran) [inline app/controllers/sessions_controller.rb]: J'ai l'impression que le `try()` est là pour protéger des cas où le paramètre serait nil. Mais si c'est le cas gsub va planter aussi, non ?

Comment (kemenaran) [inline app/controllers/sessions_controller.rb]: D'autre part je crois qu'on peut même virer le `delete(' ')`, qui est redondant avec `gsub(/\D+/, '')`.

Ce qui donnerait soit :

```ruby
params[:numero_fiscal].try(:gsub, /\D+/, '')
```

soit avec une syntaxe p-ê plus légère (mais plus de conditionnels) :

```ruby
params[:numero_fiscal] && params[:numero_fiscal].gsub(/\D+/, '')
```

File: spec/controllers/session_controller_spec.rb [hunk 2]
```diff
@@ -26,6 +26,15 @@
...
       end
 
+      context "quand mon numero fiscal se termine par une lettre" do
+        let(:numero_fiscal)  { Fakeweb::ApiParticulier::NUMERO_FISCAL.to_s + 'C' }
+
+        it "je suis redirigé vers la page de démarrage du projet" do
+          post :create, numero_fiscal: numero_fiscal, reference_avis: reference_avis, proprietaire: "1"
+          expect(response).to redirect_to etape1_recuperation_infos_path(projet)
+        end
+      end
```
Comment (kemenaran) [inline spec/controllers/session_controller_spec.rb]: Comme on a déjà un cas de succès dans les tests d'acceptance de cette étape, ça me semble un peu lourd d'en rajouter un nouveau.

ÀMHA on pourrait :

1. Soit rajouter un "C" à un test d'acceptance existant (pour rendre le test un peu plus corner-case),
2. Soit rajouter un test de contrôleur à `session_controller_spec.rb`.

J'ai une préférence pour 2 : je ne saurais pas trop expliquer pourquoi, mais j'ai l'impression que ce niveau de détail c'est plutôt le boulot d'un test de contrôleur que d'acceptance.

Comment (qcattez) [inline spec/controllers/session_controller_spec.rb]: Je comprends pas, c'est bien un test de contrôleur vu qu'on est dans `session_controller_spec.rb`

File: app/controllers/sessions_controller.rb (hunk 1 above)
Comment (qcattez) [inline app/controllers/sessions_controller.rb]: Je pencherai plus pour la première proposition qui me semble plus explicite au premier coup d'œil. Mais je comprends pas l'action qu'est sensé faire le `delete(' ')`

Comment (jvignolles) [inline app/controllers/sessions_controller.rb]: Je propose même :
```ruby
params[:numero_fiscal].to_s.gsub(/\D+/, '')
```

File: app/controllers/sessions_controller.rb [hunk 3]
```diff
@@ -47,10 +47,10 @@ def deconnexion
...
   end
 
   def param_reference_avis
-    params[:reference_avis].try(:delete, ' ')
+    params[:reference_avis].try(:delete, ' ').gsub(/\W+/, '').upcase
```
Comment (jvignolles) [inline app/controllers/sessions_controller.rb]: Je propose :
```ruby
params[:reference_avis].to_s.gsub(/\W+/, '').upcase
```

File: spec/controllers/session_controller_spec.rb (hunk 2 above)
Comment (kemenaran) [inline spec/controllers/session_controller_spec.rb]: Ah merde, comment j'ai raté ça ? Je devais pas avoir les yeux en face des trous 😅

Comment (kemenaran) [inline spec/controllers/session_controller_spec.rb]: Du coup c'est bon pour moi 👍
